tdp/core/variables/cluster_variables.py

```diff
@@ -39,11 +39,14 @@ def initialize_cluster_variables(
         Args:
             collections (Collections): instance of collections
             tdp_vars (Union[str, Path]): path to the tdp vars
-            override_folder (Optional[str | Path]): path of tdp vars overrides
+            override_folders (Iterable[str | Path]): list of path(s) of tdp vars overrides
 
         Returns:
             ClusterVariables: mapping of service with their ServiceVariables instance
         """
+        if not isinstance(override_folders, (list, tuple)):
```

Comment:
You verify the type of `override_folders`, we do not verify the type of every parameters so why verify the type for this variable ? If `override_folders` is not an `Iterable` it will fail in the `for` loop below this is fine for me.

Reply:
Since the `str` type is an iterable object, if `override_folders` is equal to `'my_vars/platform1'` and not `['my_vars/platform1']`, the first `for` loop will not fail because it will loop on every character of the string, and also because the `.iterdir()` method creates a generator without checking if the Path exists at that moment.

But in the end you are right, it will eventually fail later on at line 70 in the following `for` loop when trying to iterate over the "one character" path that probably won't exist (`for path in folder_iterator:`), giving a `FileNotFoundError`.